nixos/modules/services/cluster/spark/default.nix

```diff
@@ -0,0 +1,151 @@
+{config, pkgs, lib, ...}:
+let
+  cfg = config.services.spark;
+in
+with lib;
+{
+  options = {
+    services.spark = {
+      master = {
+        enable = mkEnableOption "Enable spark master service";
+        bind = mkOption {
+          type = types.str;
+          description = "Address the spark master binds to";
+          default = "127.0.0.1";
+          example = "0.0.0.0";
+        };
+        restartIfChanged  = mkOption {
```

Comment:
Why is someone going to change this? Is it more of an advanced feature?

Reply:
This option is very useful to have not just for this module, but for any service where a restart can have a significant impact. Some services like [vault](https://github.com/NixOS/nixpkgs/blob/nixos-21.05/nixos/modules/services/security/vault.nix#L174) have `restartIfChanged` hardcoded to false because service starts in a "locked" state and requires manual intervention to unlock. In the case of spark, if you're running a production cluster with many applications running, you might want the option to defer a restart for a more convenient time. In some other setups you might have external schedulers to automatically restart jobs that filed due to a cluster restart, or you might have a cluster with ad-hoc applications that are okay with failures.

Even within the infrastructure of my team at work we have multiple clusters that need to have restartIfChanged set to true and false. We could do this through `systemd.services.*....`, but adding the option in the module is cleaner.

Outside of this module, I think it makes sense to have this option for a few other services where automatic restarts could be expensive, such as Elasticsearch and Hadoop. I'll open an issue to discuss those later.

Reply:
If the option will be used commonly and having a good `description` provides value then I agree copying options is a reasonable idea 👍

Reply:
Oh I forgot to mention you should really highlight the security implications if this value is set to false. We want to make sure everyone understands their responsibilities if they change the value here.

Reply:
```
Automatically restart (worker/slave) service on config change.
This can be set to false to defer restarts on clusters running critical applications.
Please consider the security implications of inadvertently running an older version,
and the possibility unexpected behavior caused by inconsistent versions across a cluster when disabling this option.
```
Will this do?